core/commands/diff.py

```diff
@@ -221,18 +221,24 @@ def get_hunk_diff(self, pt):
         Given a cursor position, find and return the diff header and the
         diff for the selected hunk/file.
         """
-        header_start = self.diff_starts[bisect.bisect(self.diff_starts, pt) - 1]
-        header_end = self.diff_header_ends[bisect.bisect(self.diff_header_ends, pt) - 1]
 
-        if not header_end or header_end < header_start:
-            # The cursor is not within a hunk.
-            return
-
-        diff_start = self.hunk_starts[bisect.bisect(self.hunk_starts, pt) - 1]
-        diff_end = self.hunk_ends[bisect.bisect(self.hunk_ends, pt)]
+        for hunk_start, hunk_end in zip(self.hunk_starts, self.hunk_ends):
+            if hunk_start <= pt < hunk_end:
+                break
+        else:
+            window = self.view.window()
+            if window:
+                window.status_message('Not within a hunk')
+            return  # Error!
+
+        header_start, header_end = max(
+            (header_start, header_end)
+            for header_start, header_end in zip(self.diff_starts, self.diff_header_ends)
+            if (header_start, header_end) < (hunk_start, hunk_end)
```

Comment:
How does it work when comparing a tuple to an other tuple?
Not sure how the bisect works since it's is a long time since I looked at it

Reply:
Thanks @kaste